network-container/cmd/resource/resource.go

```diff
@@ -0,0 +1,177 @@
+	if regionName == nil || *regionName == "" {
+		return handler.ProgressEvent{}, fmt.Errorf("`error creating network container: region_name` must be set")
+	}
+	containerRequest.RegionName = *regionName
+	containerRequest.ProviderName = *providerName
+	atlasCIdRBlock := currentModel.AtlasCidrBlock.Value()
+	if atlasCIdRBlock == nil || *atlasCIdRBlock == "" {
+		return handler.ProgressEvent{}, fmt.Errorf("error creating network container: `atlasCidrBlock`")
+	}
+	containerRequest.AtlasCIDRBlock = *atlasCIdRBlock
+	containerResponse, _, err := client.Containers.Create(context.Background(), *projectID, containerRequest)
+	if err != nil {
+		return handler.ProgressEvent{}, fmt.Errorf("error creating network container: %s", err)
+	}
+
+	currentModel.Id = encoding.NewString(containerResponse.ID)
+
+	return handler.ProgressEvent{
+		OperationStatus: handler.Success,
+		Message:         "Create complete",
+		ResourceModel:   currentModel,
+	}, nil
+}
+
+// Read handles the Read event from the Cloudformation service.
+func Read(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	client, err := util.CreateMongoDBClient(*currentModel.ApiKeys.PublicKey.Value(), *currentModel.ApiKeys.PrivateKey.Value())
+
+	if err != nil {
+		return handler.ProgressEvent{}, err
+	}
+
+	projectID := *currentModel.ProjectId.Value()
+	containerID := *currentModel.Id.Value()
+
+	containerResponse, _, err := client.Containers.Get(context.Background(), projectID, containerID)
+
+	if err != nil {
+		return handler.ProgressEvent{}, fmt.Errorf("error reading container with id(project: %s, container: %s): %s", projectID, containerID, err)
+	}
+
+	currentModel.RegionName = encoding.NewString(containerResponse.RegionName)
+	currentModel.Provisioned = encoding.NewBool(*containerResponse.Provisioned)
+	currentModel.VpcId = encoding.NewString(containerResponse.VPCID)
+	currentModel.ContainerId = encoding.NewString(containerResponse.ID)
+	currentModel.AtlasCidrBlock = encoding.NewString(containerResponse.AtlasCIDRBlock)
+
+	return handler.ProgressEvent{
+		OperationStatus: handler.Success,
+		Message:         "Read Complete",
+		ResourceModel:   currentModel,
+	}, nil
+}
+
+// Update handles the Update event from the Cloudformation service.
+func Update(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	client, err := util.CreateMongoDBClient(*currentModel.ApiKeys.PublicKey.Value(), *currentModel.ApiKeys.PrivateKey.Value())
+
+	if err != nil {
+		return handler.ProgressEvent{}, err
+	}
+
+	projectId := *currentModel.ProjectId.Value()
+	containerId := *currentModel.Id.Value()
+	containerRequest := &matlasClient.Container{}
+	providerName := currentModel.ProviderName.Value()
+	atlasBlock := currentModel.AtlasCidrBlock.Value()
+	if atlasBlock != nil {
+		containerRequest.AtlasCIDRBlock = *atlasBlock
+	}
+	if providerName == nil || *providerName == "" {
+		aws := defaultProviderName
+		providerName = &aws
+	}
```

Comment:
Let's move this block above `atlasBlock := currentModel.AtlasCidrBlock.Value()` so that we're doing nil checking immediately after declaring the variable.